core/src/test/scala/org/apache/spark/internal/config/ConfigEntrySuite.scala

```diff
@@ -289,6 +289,70 @@ class ConfigEntrySuite extends SparkFunSuite {
     assert(conf.get(iConf) === 3)
   }
 
+  test("conf entry: prepend with default separator") {
```

Comment:
Should probably add a test for the "fallback conf" case, since it's a little odd. If I follow the current code, it will not fallback if either the conf or the "prepend" conf are set (which is fine, it just would be good to have a test).

Reply:
Yep, fallback is a missing spot, handling it...